src/main/java/uk/gov/companieshouse/company/metrics/exception/RetryableTopicErrorInterceptor.java

```diff
@@ -0,0 +1,52 @@
+package uk.gov.companieshouse.company.metrics.exception;
+
+import static java.lang.String.format;
+import static org.springframework.kafka.support.KafkaHeaders.EXCEPTION_CAUSE_FQCN;
+import static org.springframework.kafka.support.KafkaHeaders.EXCEPTION_STACKTRACE;
+
+import java.util.Map;
+
+import org.apache.kafka.clients.producer.ProducerInterceptor;
+import org.apache.kafka.clients.producer.ProducerRecord;
+import org.apache.kafka.clients.producer.RecordMetadata;
+import org.apache.kafka.common.header.Header;
+import uk.gov.companieshouse.company.metrics.config.LoggingConfig;
+
+public class RetryableTopicErrorInterceptor implements ProducerInterceptor<String, Object> {
+
+    @Override
+    public ProducerRecord<String, Object> onSend(ProducerRecord<String, Object> record) {
+        String nextTopic = record.topic().contains("-error") ? getNextErrorTopic(record)
+                : record.topic();
+        LoggingConfig.getLogger().info(format("Moving record into new topic: %s with value: %s",
+                nextTopic, record.value()));
+        if (nextTopic.contains("-invalid")) {
```

Comment:
Can you please integrate Logger and log the topic its forwarding it, have a look at charges delta consumer for reference

Reply:
This has already done and is inline with charges delta consumer.